### PR TITLE
feat: todo feature

### PR DESCRIPTION
**Pull request by deepak29-git**
Added todo feature

Screenshot (144)

### CODE REVIEW

File: src/Components/TodoModal/TodoModal.jsx [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+            {todoInput && (
+              <input
+                onKeyDown={(e) => {
+                  const addObj = {
+                    id: uuid(),
+                    name: input,
+                    isDone: false,
+                  };
+                  e.key === "Enter" && setTodo([...todo, addObj]);
+                  // setInput("");
```
Comment (tushargoyal812) [inline src/Components/TodoModal/TodoModal.jsx:63]: remove comments

File: src/Components/TodoModal/TodoModal.jsx [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+                }}
+                onChange={(e) => setInput(e.target.value)}
+                className="todo-input"
+                placeholder="New Todo"
+                type="text"
+                // value={input}
```
Comment (tushargoyal812) [inline src/Components/TodoModal/TodoModal.jsx:69]: remove comments

**Review by tushargoyal812 — COMMENTED** [review]
code is clean and to the point.
variable naming is good.
file structure is good.

File: src/Components/Todo/Todo.css [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+  margin-right: 1rem;
+  position: relative;
+  cursor: pointer;
+}
+
+.todo-title::before {
+  content:attr(data-hover);
+  visibility: hidden;
+  position: absolute;
+  bottom: 30px;
```
Comment (Mohitkumar33) [inline src/Components/Todo/Todo.css:17]: You can use rems. here

File: src/Components/Todo/Todo.jsx [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+  const todoHandler = () => {
+    if (todoModal) {
+      setTodoModal(false);
+    } else {
+      setTodoModal(!todoModal);
+    }
+  };
+  return (
+    <div className="align-end">
+      {todoModal && <TodoModal />}
```
Comment (Mohitkumar33) [inline src/Components/Todo/Todo.jsx:15]: You can use ? : because this syntax gives error in console

File: src/Components/TodoModal/TodoModal.css [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+  display: flex;
+  justify-content: center;
+  align-items: center;
+  height: 80%;
+  width: 100%;
+  flex-direction: column;
+}
+
+.add-todo-btn {
+  background-color: var(--primary-color);
```
Comment (Mohitkumar33) [inline src/Components/TodoModal/TodoModal.css:20]: nice use of color variables

File: src/Components/TodoModal/TodoModal.jsx [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+                      todo.isDone ? `done-todo` : undefined
+                    }`}
+                  >
+                    {todo.name}
+                  </label>
+                </div>
+              ))
+            )}
+
+            {!todoInput && (
```
Comment (Mohitkumar33) [inline src/Components/TodoModal/TodoModal.jsx:44]: you can use ? : as the syntax you are using may give error in console

File: src/Components/TodoModal/TodoModal.jsx (hunk 1 above)
Comment (Mohitkumar33) [inline src/Components/TodoModal/TodoModal.jsx:63]: remove commented code

**Review by Mohitkumar33 — APPROVED** [review]
Nice Code. Need to do some changes

File: src/Components/Todo/Todo.css (hunk 3 above)
Comment (deepak29-git) [reply src/Components/Todo/Todo.css:17]: ok

File: src/Components/Todo/Todo.jsx (hunk 4 above)
Comment (deepak29-git) [reply src/Components/Todo/Todo.jsx:15]: ok

File: src/Components/TodoModal/TodoModal.jsx (hunk 1 above)
Comment (deepak29-git) [reply src/Components/TodoModal/TodoModal.jsx:63]: ok

**Pull request merged**